numpy/core/tests/test_numeric.py

```diff
@@ -140,6 +140,20 @@ def test_round(self):
         tgt = [1.6, 72.5, 6.4, 3.2]
         assert_equal(np.around(arr, decimals=1), tgt)
 
+        s = np.float64(1.)
+        assert_(isinstance(round(s), int))
+        assert_(isinstance(s.round(), np.float64))
+        assert_equal(s.round(), 1.)
+        assert_equal(round(s), 1)
+
+        s = np.uint64(2**64-1)
+        assert_equal(round(s), 2**64-1)
+
+        f = np.float64(5.1 * 10**73)
+        assert_(isinstance(round(f, -73), float))
+        assert_allclose(round(f, -73), 5.0 * 10**73)
```

Comment:
Perhaps better to just test directly against the python behavior
```suggestion
        assert_allclose(round(f, -73), round(float(f), -73))
```

Reply:
Doesn't always work unfortunately.

```python
>>> round(f, -75) == round(float(f), -73)
False
```

We would need to change the implementation significantly, depending on the dtype.

Reply:
Did you try using 73 on both sides and not just on the right?

Reply:
```python
>>> round(f, -73) == round(float(f), -73)
False
```

Reply:
Can you add an `xfail` test that checks for consistency against python then?

Reply:
Done.